Setup Tests for Components

milesmace opened the pull request:
## Description:
- Add tests for components
- Covered `utils`, `ui`, `auth`, `layout` components in testing

milesmace committed:
test: set up testing with `vitest`

milesmace committed:
fix: Update vite config for better path resolution

milesmace committed:
test: Add test for `ui` & `utils` components

milesmace committed:
testing: Add tests for `withProtected` & `withRestrictedPublic`

milesmace committed:
chore: Check if code is linted & formatted

what's changed:
- Check if the code is linted & formatted rather than linting and
  formatting

milesmace committed:
test: Set up tests for `NavBar` component

milesmace committed:
chore: Use route constants instead of hard-coded routes in tests

milesmace committed:
Merge pull request #10 from zenitsua2416/chore/husky

Fix Husky Setup Error

krishnamodepalli commented on client/src/components/auth/withProtected/withProtected.spec.tsx:
```diff
@@ -0,0 +1,93 @@
+
+const mockedUseAppSelector = hooks.useAppSelector as unknown as Mock;
+
+const DummyComponent = () => <>SECRET content in here!!!</>;
+const LoginComponent = () => <>Login Page</>;
+const FallBackComponent = () => <>FALLBACK</>;
```
Update as strings and use them in the queryByText.

krishnamodepalli commented on client/src/components/auth/withRestrictedPublic/withRestrictedPublic.spec.tsx:
```diff
@@ -0,0 +1,96 @@
+        <MyRoutes restrictedComponent={<PublicRestrictedComponent />} />
+      </MemoryRouter>,
+    );
+
```
Update this as in `NavBar.spec.tsx`, using a setupWithRestrictedPublic function. Check all places.

krishnamodepalli commented on client/src/components/ui/Note/Note.spec.tsx:
```diff
@@ -0,0 +1,34 @@
+    created_at: new Date().toISOString(),
+  };
+
+  it("renders the note title and link correctly", () => {
```
Add a simple test to check for description.

milesmace committed:
test(components): Improve component testing

what's changed:
- Remove boilerplate for each test with a file level global setup function
- Resovles #12 comments
